Minio.Examples/Program.cs

```diff
@@ -112,11 +112,11 @@ public static async Task Main()
         var lockBucketName = GetRandomName();
         var progress = new Progress<ProgressReport>(progressReport =>
         {
-            Console.WriteLine(
-                $"Percentage: {progressReport.Percentage}% TotalBytesTransferred: {progressReport.TotalBytesTransferred} bytes");
-            if (progressReport.Percentage != 100)
-                Console.SetCursorPosition(0, Console.CursorTop - 1);
-            else Console.WriteLine();
+            // Console.WriteLine(
+            //     $"Percentage: {progressReport.Percentage}% TotalBytesTransferred: {progressReport.TotalBytesTransferred} bytes");
+            // if (progressReport.Percentage != 100)
+            //     Console.SetCursorPosition(0, Console.CursorTop - 1);
+            // else Console.WriteLine();
```

Comment:
@ramondeklein,

> I'm not a fan of this extended behaviour of including metadata in object listings, ...

MdinIO SDKs support including object metadata when listing objects and it is a MinIO specific feature, meaning AWS S3 does not support, but we do.
So, I don't think we have a saying whether it needs to be included or not at this point.

I am totally fine replacing the SDK completely by redesigning/re-architecting it and current if decided, however until that decision is made, we obviously have to support the current SDK.

Reply:
It looks like the Go SDK supports returning metadata if I look at this code:

https://github.com/minio/minio-go/blob/95db455478e33a69eb682dff9772494228695cb9/api-list.go#L540-L542

But that works quite different then the solution that is implemented here. It seems that you can add `metadata=true` as a query parameter when calling the MinIO API and it returns the metadata in the response (so no additional roundtrips are needed). The implementation that is provided in this PR works different. It will use the normal LIstObjects method and it will call `StatObjectAsync` for each returned object. The advantage is that it also works on AWS S3, but the drawback is that it is introducing the 1+N call problem that will result in terrible performance.

Sorry for being so critical in this PR, but I think this PR shouldn't be merged this way. If we implement it, then we should use the `metadata=true` query parameter and be able to deserialize the metadata from the ListObjects response instead.

Reply:
> Not sure if you meant to disable this.

This was actually added and kept as an example, but I realize now that it causes confusion, so I'll remove them.
Done.

Reply:
> add metadata=true as a query parameter when calling the MinIO API and it returns the metadata in the response 

Totally agree.
I'll start replacing the existing logic and implement the logic in golang way.